modules/security/pam.nix

```diff
@@ -7,63 +7,80 @@ let
 
   # Implementation Notes
   #
-  # We don't use `environment.etc` because this would require that the user manually delete
-  # `/etc/pam.d/sudo` which seems unwise given that applying the nix-darwin configuration requires
-  # sudo. We also can't use `system.patchs` since it only runs once, and so won't patch in the
-  # changes again after OS updates (which remove modifications to this file).
-  #
-  # As such, we resort to line addition/deletion in place using `sed`. We add a comment to the
-  # added line that includes the name of the option, to make it easier to identify the line that
-  # should be deleted when the option is disabled.
-  mkSudoTouchIdAuthScript = isEnabled:
+  # Uses `environment.etc` to create the `/etc/pam.d/sudo_local` file that will be used
+  # to manage all things pam related for nix-darwin. An activation script will run to check
+  # for the existance of the line `auth       include        sudo_local`. This is included
+  # in macOS Sonoma and later. If the line is not there already then `sed` will add it.
+  # In those cases, the line will include the name of the option root (`security.pam`),
+  # to make it easier to identify the line that should be deleted when the option is disabled.
+  mkIncludeSudoLocalScript = isEnabled:
   let
-    file   = "/etc/pam.d/sudo";
-    option = "security.pam.enableSudoTouchIdAuth";
+    file = "/etc/pam.d/sudo";
+    option = "security.pam";
```

Comment:
```suggestion
    marker = "security.pam.sudo_local";
```

Reply:
The intention of using the root option name as the marker was to catch both the new option and old option in the one `if` condition on line 35 for when the option is disabled. This is to prevent the edge case where someone has evaluated the config, upgraded to the new version, and then disabled the `security.pam.enableSudoTouchIdAuth` option before evaluating it again. We need to remove any line containing `security.pam` as a precaution against this.